scripts/js/commands/checkKatexRender.ts

```diff
@@ -0,0 +1,269 @@
+    );
+    process.exit(1);
+  }
+
+  console.log("\nNo errors found ✅\n");
+};
+
+async function checkPage(page: Page, pathname: string): Promise<string[]> {
+  const errors: string[] = [];
+  try {
+    await page.goto(`http://localhost:3000/${pathname}`);
+
+    const allKatex = page.locator("span.katex");
+    const count = await allKatex.count();
+
+    for (let i = 0; i < count; i++) {
+      const mathExpr = allKatex.nth(i);
+
+      // Expressions that are not inline have a parent span with the `katex-display` class.
+      const isKatexDisplay = await mathExpr.evaluate((expr: any) =>
+        expr.parentElement?.classList.contains("katex-display"),
+      );
+
+      // We skip math expressions that are not inline
+      if (isKatexDisplay) {
+        continue;
+      }
+
+      const box = await mathExpr.boundingBox();
+      if (box && box.width >= MIN_WIDTH) {
+        const content = await mathExpr.textContent();
+        errors.push(
+          `  -  Error found. Approximate preview of the expression:\n\t $${content}}$`,
+        );
+      }
+    }
+  } catch (_) {
+    errors.push("The page didn't load correctly");
+  }
+  return errors;
+}
+
+async function getRawGlobs(args: Arguments): Promise<string[]> {
+  if (args.fromFile) {
+    const content = await fs.readFile(args.fromFile, "utf-8");
+    return globby(content.split("\n").filter((entry) => entry));
+  }
+
+  const globs = [];
+  if (args.learning) {
+    globs.push("learning/**/*.{ipynb,mdx}");
+  }
+
+  if (args.nonApiDocs) {
+    globs.push("docs/**/*.{ipynb,mdx}");
+  }
+
+  for (const [isIncluded, glob] of [
+    [args.currentApis, `docs/api/*/*.mdx`],
+    [args.historicalApis, "docs/api/*/[0-9]*/*.mdx"],
+    [args.devApis, "docs/api/*/dev/*.mdx"],
+    [args.qiskitReleaseNotes, "docs/api/qiskit/release-notes/*.mdx"],
+    ...IGNORED_FILES_SHOULD_FIX.map((file) => [args.ignoredFiles, file]),
+  ]) {
+    const prefix = isIncluded ? "" : "!";
+    globs.push(`${prefix}${glob}`);
+  }
+
+  return globby(globs);
+}
+
+export function hasInlineMath(markdown: string): boolean {
+  return !!markdown.match(/(?<!\$)\$(?!\$)(.+?)(?<!\$)\$(?!\$)/);
+}
+
+export function removeFileExtension(pathname: string): string {
+  const extensions = [".mdx", ".ipynb"];
+
+  for (const extension of extensions) {
+    if (pathname.endsWith(extension)) {
+      return removeSuffix(pathname, extension);
+    }
+  }
+
+  return pathname;
+}
+
+async function determineFilePaths(args: Arguments): Promise<string[]> {
+  const globs = await getRawGlobs(args);
+
+  const filesWithMath = [];
+  for (const glob of globs) {
+    const markdown = await readMarkdown(glob);
+    // Only get the files that contain inline math expressions
+    if (hasInlineMath(markdown)) {
+      filesWithMath.push(glob);
+    }
+  }
+
+  return filesWithMath.map((f) =>
+    // The pages must be accesses without the file extensions
+    removeFileExtension(f),
+  );
+}
+
+main().then(() => process.exit());
```

Comment:
I would expect this to mess up the tests because importing the file runs this code. This is why Python has `if __name__ == "__main__"`. You can see similar code we have in the closed source infra repository for redirects. Or, the approach we normally take in this repo is to move the helper functions to `lib/`